packages/menu/src/SubMenu.tsx

```diff
@@ -0,0 +1,349 @@
+import React, { RefObject, useRef } from 'react';
+import PropTypes from 'prop-types';
+import { Transition } from 'react-transition-group';
+import IconButton from '@leafygreen-ui/icon-button';
+import Icon, { glyphs } from '@leafygreen-ui/icon';
+import { css, cx } from '@leafygreen-ui/emotion';
+import { uiColors } from '@leafygreen-ui/palette';
+import { HTMLElementProps, createDataProp } from '@leafygreen-ui/lib';
+import { useUsingKeyboardContext } from '@leafygreen-ui/leafygreen-provider';
+import {
+  menuItemContainerStyle,
+  activeMenuItemContainerStyle,
+  disabledMenuItemContainerStyle,
+  focusedMenuItemContainerStyle,
+  linkStyle,
+  disabledTextStyle,
+} from './styles';
+import { ExitHandler } from 'react-transition-group/Transition';
+
+const subMenuContainer = createDataProp('sub-menu-container');
+
+const liStyle = css`
+  position: relative;
+  overflow: hidden;
+`;
+
+const subMenuStyle = css`
+  flex-direction: row;
+  min-height: 56px;
+  border-top: 1px solid ${uiColors.gray.light2};
+  background-color: ${uiColors.gray.light3};
+  align-items: center;
+  justify-content: flex-start;
+`;
+
+const subMenuOpenStyle = css`
+  background-color: ${uiColors.white};
+  &:hover {
+    background-color: ${uiColors.gray.light2};
+  }
+`;
+
+const closedIconStyle = css`
+  transition: color 200ms ease-in-out;
+  color: ${uiColors.gray.base};
+`;
+
+const openIconStyle = css`
+  color: ${uiColors.gray.dark2};
+`;
+
+const mainTextStyle = css`
+  display: flex;
+  flex-direction: row;
+  justify-content: space-between;
+  font-size: 14px;
+  color: ${uiColors.gray.dark3};
+  ${subMenuContainer.selector}:focus & {
+    color: ${uiColors.blue.dark3};
+  }
+`;
+
+const activeMainTextStyle = css`
+  font-weight: bolder;
+  color: ${uiColors.green.dark3};
+`;
+
+const iconButtonStyle = css`
+  position: absolute;
+  z-index: 1;
+  right: 8px;
+  top: ${52 / 2 - 22 / 2}px;
```

Comment:
Can you comment these magic numbers, or make them variables?

Reply:
👍